### PR TITLE
Updated dependecies, added useEffect support

### LINKED ISSUE #88: React hooks & context
I'd love to be able to use [react hooks](https://reactjs.org/hooks) in react-blessed.

It's an RFC, now, but looks like it will probably solidify into mainline react, soon. In order to test, I installed `react@16.7.0-alpha.2`.

As it is, I get this error:

```
Invariant Violation: Hooks can only be called inside the body of a function component.
```

with this code:

```js
import React, { useState, useEffect } from 'react'
import fetch from 'node-fetch'

function App () {
  const [posts, setPosts] = useState([])
  useEffect(async () => {
    const items = await fetch(
      'https://jsonplaceholder.typicode.com/posts'
    ).then(r => r.json())
    setPosts(items.map(i => i.title))
  })

  return (
    <list items={posts} />
  )
}
```

To verify that the hooks work, etc, [here](https://codesandbox.io/s/py6r7xq6z0) is a similar thing in a sandbox, in react-dom.

### PR DESCRIPTION
**Pull request by tomaskallup**
I made some small changes, mainly updating dependecies, making sure hooks work on the stable react version.
Also I tried my best to implement passive effects (`useEffect`), they seem to work now, but it's hard to tell if I did it right considering the documentation around react-reconciler.
I believe this should finally fix #88, altough there are still some untested hooks:
- useCallback
- useMemo
- useRef
- useImperativeHandle
- useLayoutEffect (this one might need some actual reconciler implementation)
- useDebugValue

### CODE REVIEW

**Comment by Yomguithereal** [comment]
Hello @tomaskallup. Thanks for your PR. Can I ask you whether you think there is anything breaking in your addenda?

**Comment by tomaskallup** [comment]
Hi @Yomguithereal, there shouldn't be anything breaking.
We already had hooks working - those are now updated to stable react version - and effect were never working.

File: src/fiber/fiber.js [hunk 1]
```diff
@@ -203,6 +204,18 @@ const createBlessedRenderer = function(blessed) {
...
+    schedulePassiveEffects(effect) {
+      effect();
+
+      runningEffects.push(effect);
+    },
+
+    cancelPassiveEffects() {
+      runningEffects.forEach(effect => effect.cancel());
+
+      runningEffects.splice(0, runningEffects.length);
```
Comment (Yomguithereal) [inline src/fiber/fiber.js]: Wouldn't a `runningEffects.length = 0` or a reset to a new `[]` be better and more legible?

Comment (tomaskallup) [reply src/fiber/fiber.js]: Probably would be, fixed.

File: package.json [hunk 2]
```diff
@@ -51,11 +51,11 @@
   },
   "peerDependencies": {
     "blessed": "^0.1.81",
-    "react": "^16.6.3",
     "react-devtools-core": "^3.0.0",
     "ws": "^6.0.0"
   },
   "dependencies": {
-    "react-reconciler": "^0.18.0-alpha.2"
+    "react": "^16.0.0",
```
Comment (Yomguithereal) [inline package.json]: Why is `react` both in deps and dev deps?

Comment (tomaskallup) [reply package.json]: It was like that before, I felt like it was wrong, resolved.

**Comment by Yomguithereal** [comment]
Would you release a v0.6.0 or a v0.5.1?

**Comment by tomaskallup** [comment]
I would say v0.6.0, as this introduces some new features and it's backwards-compatible.

**Comment by Yomguithereal** [comment]
Done. v0.6.0 is now live on npm.

**Comment by Yomguithereal** [comment]
> as this introduces some new features and it's backwards-compatible.

I think I remember however that versions < 1.0.0 are a bit quirky in npm as minors are considered majors when automatically bumping version at install time.

**Comment by tomaskallup** [comment]
https://semver.org/#spec-item-4
It says that anything can break at any point for the major < 1 version. But there is also this:

> Minor version Y (x.Y.z | x > 0) MUST be incremented if new, backwards compatible functionality is introduced to the public API. It MUST be incremented if any public API functionality is marked as deprecated. It MAY be incremented if substantial new functionality or improvements are introduced within the private code. It MAY include patch level changes. Patch version MUST be reset to 0 when minor version is incremented.

**Pull request merged**